open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+                return validationResult;
+            }
+
+        } else {
+            log.error(ErrorConstants.PAYLOAD_FORMAT_ERROR_CREDITOR_ACC);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validates the presence of a specified key in a JSONObject (either the amount or the currency)
+     * and checks if the associated value is a non-empty string.
+     *
+     * @param parentObj The JSONObject to be validated.
+     * @param key       The key to be checked for presence in the parentObj.
+     * @return true if the specified key is present in the parentObj and the associated value is a
+     * non-empty string.
+     */
+    public static JSONObject validateJsonObjectKey(JSONObject parentObj, String key) {
+        JSONObject validationResponse = new JSONObject();
+
+        if (parentObj != null) {
+            // Check if the specified key is present in the parentObj
+            if (parentObj.containsKey(key)) {
+                Object value = parentObj.get(key);
+
+                // Check if the value associated with the key is a non-empty string
+                if (value instanceof String && !((String) value).isEmpty()) {
+                    validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+                    return validationResponse; // Valid: The key is present, and the value is a non-empty String
+                } else {
+                    String errorMessage = "The value of '" + key + "'is not a string or the value is empty";
+                    return ConsentManageUtil.getValidationResponse(errorMessage);
+                    // Invalid: The value associated with the key is not a non-empty String
+                }
+            } else {
+                String errorMessage = "Mandatory parameter '" + key + "' is not present in payload";
+                return ConsentManageUtil.getValidationResponse(errorMessage);
+                // Invalid: The specified key is not present in parentObj
+            }
+        }
+
+        String errorMessage = "parameter passed in  is null";
+        return ConsentManageUtil.getValidationResponse(errorMessage);
+    }
+
+    /**
+     * Validates the presence of a specified key in a JSONArray (either the amount or the currency)
+     * in periodiclimits and checks if the associated value is a non-empty string.
+     *
+     * @param parentArray The JSONObject to be validated.
+     * @param key         The key to be checked for presence in the parentObj.
+     * @return true if the specified key is present in the parentObj and the associated value is a
+     * non-empty string.
+     */
+
+    public static JSONObject validateAmountCurrencyPeriodicLimits(JSONArray parentArray, String key) {
```

Comment:
Shall we make this method generic too? rather than for only amount or the currency?

Reply:
Also we can reuse the logic in `validateJsonObjectKey` (above method) for validation of the elements inside array.

Reply:
Fixed in [9ab5abf]